pkg/kapp/clusterapply/waiting_changes.go

```diff
@@ -100,6 +105,13 @@ func (c *WaitingChanges) WaitForAny() ([]WaitingChange, error) {
 			case !state.Done:
 				newInProgressChanges = append(newInProgressChanges, change)
 
+				if _, found := change.Cluster.Resource().Annotations()[nonBlockingOrderingAnnKey]; found {
+					// TODO Improve if condition
+					if !strings.Contains(state.Message, "Waiting for generation") {
```

Comment:
i dont think we need to check on any info here. if resource is marked as "wait is non-blocking" then we should unblock other changes, but also keep waiting for change to finish.

Reply:
> Waiting for the resource to be created/updated on the api may be sufficient, or if a wait rule defines supportsObservedGeneration: true wait for the generations to match.

The check is inline with this ^. I am also wondering if we should allow a value for the annotation to make this check opt in 🤔

Reply:
Removed the check, but keeping this open for now.